cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -174,10 +175,14 @@ public BodyConsumer createAppVersion(HttpRequest request, HttpResponder responde
                                          @PathParam("namespace-id") final String namespaceId,
                                          @PathParam("app-id") final String appId,
                                          @PathParam("version-id") final String versionId)
-    throws BadRequestException, NamespaceNotFoundException {
+    throws Exception {
 
     ApplicationId applicationId = validateApplicationVersionId(namespaceId, appId, versionId);
 
+    if (!applicationLifecycleService.updateAppAllowed(applicationId))  {
+      responder.sendString(HttpResponseStatus.INTERNAL_SERVER_ERROR, "There already exists application version " +
```

Comment:
This is not an internal server error since user is trying to add multiple version of the same app. I think conflict would be better.